datalad_fuse/fuse_.py

```diff
@@ -314,3 +337,20 @@ def file_getattr(f):
     data["st_ctime"] = time.time()
     data["st_mtime"] = time.time()
     return data
+
+
+
+# might be called twice in rapid succession for an annex key path
+@lru_cache(maxsize=CACHE_SIZE)
+def is_annnex_dir_or_key(path: str):
```

Comment:
It would be better to operate on `Path(path).parts` instead of indexing a string.  For one thing, then you wouldn't get false-positives with directories named "foo.git/annex".